sycl/doc/design/OffloadDesign.md

```diff
@@ -0,0 +1,396 @@
+# Implementation design for offloading model
+
+## Introduction
+
+This document covers the implementation design for using the new offloading
+model for the DPC++ Compiler.  This leverages the existing community Offloading 
+design [OffloadingDesign][1] which covers the Clang driver and code generation
+steps for creating offloading applications.
+
+[1]: <../../../clang/docs/OffloadingDesign.rst>
+
+The current offloading model is completely encapsulated within the Clang
+Compiler Driver requiring the driver to perform all of the additional steps
+for generating the host and device compilation passes.  The Driver is also
+responsible for performing any of the link-time processing that occurs for
+each device target.
+
+The updated offloading model removes much of the functionality that is
+performed during the link phase of the offloading compilation and moves it
+to a `clang-linker-wrapper` tool.
+
+Below is a general representation of the overall offloading flow that is
+performed during a full compilation from source to final executable.  The
+compiler driver is responsible for creating the fat object and the
+`clang-linker-wrapper` tool is responsible for the general functionality that is
+performed during the link.
+
+![High level view of the offloading flow](images/OffloadGeneralFlow.svg
+ "General Offload Flow")
+
+*Diagram 1: General Offload Flow*
+
+More specific details of the clang-linker-wrapper behavior can be seen in the
+[clang-linker-wrapper](#clang-linker-wrapper) section.
+
+## Fat Binary Representation
+
+Binaries generated during the offload compilation will be 'bundled' together
```

Comment:
Currently, this sub-section seems low on information. I suppose there is more info in the community 'offloading design' document? May be a link to that information can be useful. A modified version of that information embedded here may be even better.

Thanks